config/functions

```diff
@@ -1,13 +1,31 @@
 setup_toolchain() {
-  # lto flag
-  if flag_enabled "lto" "$LTO_SUPPORT" "only-disable"; then
-    TARGET_CFLAGS+=" $CFLAGS_OPTIM_LTO"
-    TARGET_CXXFLAGS+=" $CXXFLAGS_OPTIM_LTO"
-    TARGET_LDFLAGS+=" $LDFLAGS_OPTIM_LTO"
-  else
-    TARGET_CFLAGS+=" $CFLAGS_OPTIM_NOLTO"
-    TARGET_CXXFLAGS+=" $CXXFLAGS_OPTIM_NOLTO"
-    TARGET_LDFLAGS+=" $LDFLAGS_OPTIM_NOLTO"
+  TARGET_AR_NM_RANLIB_PREFIX=""
+
+  if [ "$LTO_SUPPORT" = "yes" ]; then
+    if flag_enabled "lto-parallel" "no"; then
+      TARGET_CFLAGS+=" $FLAGS_OPTIM_LTO_PARALLEL $FLAGS_OPTIM_LTO_NO_FAT"
+      TARGET_CXXFLAGS+=" $FLAGS_OPTIM_LTO_PARALLEL $FLAGS_OPTIM_LTO_NO_FAT"
+      TARGET_LDFLAGS+=" $LDFLAGS_OPTIM_LTO_COMMON $FLAGS_OPTIM_LTO_PARALLEL"
+      # static libs with lto info require gcc-ar/gcc-nm/gcc-ranlib instead of
+      # default binutils versions
+      TARGET_AR_NM_RANLIB_PREFIX="gcc-"
+    elif flag_enabled "lto-fat" "no"; then
+      TARGET_CFLAGS+=" $FLAGS_OPTIM_LTO_NO_PARALLEL $FLAGS_OPTIM_LTO_FAT"
+      TARGET_CXXFLAGS+=" $FLAGS_OPTIM_LTO_NO_PARALLEL $FLAGS_OPTIM_LTO_FAT"
+      TARGET_LDFLAGS+=" $LDFLAGS_OPTIM_LTO_COMMON $FLAGS_OPTIM_LTO_NO_PARALLEL"
+      TARGET_AR_NM_RANLIB_PREFIX="gcc-"
+    elif flag_enabled "lto" "no"; then
+      TARGET_CFLAGS+=" $FLAGS_OPTIM_LTO_NO_PARALLEL $FLAGS_OPTIM_LTO_NO_FAT"
+      TARGET_CXXFLAGS+=" $FLAGS_OPTIM_LTO_NO_PARALLEL $FLAGS_OPTIM_LTO_NO_FAT"
+      TARGET_LDFLAGS+=" $LDFLAGS_OPTIM_LTO_COMMON $FLAGS_OPTIM_LTO_NO_PARALLEL"
+      TARGET_AR_NM_RANLIB_PREFIX="gcc-"
+    fi
+  fi
+
+  if flag_enabled "lto-off" "no"; then
```

Comment:
Should this be:
```
if [ "$LTO_SUPPORT" != "yes" ] || flag_enabled "lto-off" "no"; then
```
?

Reply:
The idea behind this was to not set any lto flags unless they are requested by build flags. So packages like kodi can set their own flags and we don't run into the problem of possibly contradicting flags.

Enabling LTO is guarded by LTO_SUPPORT, but for lto-off I think it's better to always disable it whenever a package requests it.

Reply:
OK, thanks.